### PR TITLE
resource/aws_cloudwatch_log_metric_filter: Add support for DefaultValue

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #1563
Fixes: #1442

```
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSCloudWatchLogMetricFilter_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSCloudWatchLogMetricFilter_ -timeout 120m
=== RUN   TestAccAWSCloudWatchLogMetricFilter_basic
--- PASS: TestAccAWSCloudWatchLogMetricFilter_basic (59.02s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	59.047s
```

### CODE REVIEW

File: aws/structure.go [hunk 1]
```diff
@@ -1451,14 +1451,30 @@ func expandApiGatewayStageKeyOperations(d *schema.ResourceData) []*apigateway.Pa
...
 
-	return []*cloudwatchlogs.MetricTransformation{&transformation}
+	if m["default_value"] != "" {
+		bound := m["default_value"]
+		switch bound := bound.(type) {
+		case string:
+			f, err := strconv.ParseFloat(bound, 64)
+			if err != nil {
+				return nil, fmt.Errorf(
+					"default_value must be a float value represented as a string")
```
Comment (Ninir) [inline aws/structure.go]: Do you think it would be worth to make a validation function for that?

Comment (stack72) [reply aws/structure.go]: @Ninir I changed to use the built in schema.TypeFloat to avoid the need for parsing

**Comment by stack72** [comment]
Hi @Ninir 

I believe this is ready to go now, I removed the need for the Parsing of the Double and made the schema use schema.TypeDouble instead

Paul

**Comment by Ninir** [comment]
Hi Paul,

Just checked and it looks very good. However, when updating the default value from the AWS console, the terraform plan does not see the change.

Would you have time to investigate?

Thanks! 👍

**Comment by stack72** [comment]
@Ninir I don't have access to an AWS console to test this I am afraid

Can we follow up with this change in a different PR?

**Comment by Ninir** [comment]
Hey @stack72 

This will lead to expected issues from users, so I would prefer to get the case fixed at the time! 😅 
If we are sure to get it in the same (next) release, then sure!

**Comment by endemics** [comment]
Hi @stack72 @Ninir is there any way for me to help get this accepted/merged?

**Comment by Ninir** [comment]
Hi @endemics Just started the review! i'm fixing the [update from console issue](https://github.com/terraform-providers/terraform-provider-aws/pull/2086) in a first time :)

**Comment by endemics** [comment]
Thanks @Ninir, much appreciated!

**Pull request merged**